server/grpc_service.go

```diff
@@ -1796,3 +1801,40 @@ func (s *GrpcServer) sendAllGlobalConfig(ctx context.Context, server pdpb.PD_Wat
 func (s *GrpcServer) ReportBuckets(pdpb.PD_ReportBucketsServer) error {
 	panic("not implemented")
 }
+
+// Evict the leaders when the store is damaged. Damaged regions are emergency errors
+// and requires user to manually remove the `evict-leader-scheduler` with pd-ctl
+func (s *GrpcServer) handleDamagedStore(stats *pdpb.StoreStats) error {
+	// TODO: regions have no special process for the time being
+	// and need to be removed in the future
+	damagedRegions := stats.GetDamagedRegionsId()
+	if len(damagedRegions) == 0 {
+		return nil
+	}
+
+	log.Error("store damaged and leaders will be evicted, you might fix the store and remove evict-leader-scheduler manually",
```

Comment:
Will it be printed for every store heartbeat?

Reply:
Preferably, my original intention is to warn users by all means possible to deal with this problem immediately. In the tikv part, there are critical error metric and log errors. If the user does not deal with that, damaged tikv will panic 24hours after the problem is first discovered.